emailproxy.py

```diff
@@ -574,6 +574,31 @@ class OAuth2Helper:
     class TokenRefreshError(Exception):
         pass
 
+    @staticmethod
+    def get_encryption_info(config, username, password):
+        """Returns a tuple of the fernet encrypter/decrypter and the random salt used in base64. This fernet is used to
+        store sensitive information for this account, including any stored tokens."""
+        token_salt = config.get(username, 'token_salt', fallback=None)
+        decoded_salt = None
+
+        # try to base64 decode the existing salt
+        if token_salt:
+            try:
+                decoded_salt = base64.b64decode(token_salt.encode('ascii'))  # catch incorrect third-party proxy guide
+            except (binascii.Error, UnicodeError):
+                Log.info('%s: Invalid `token_salt` value found in config file entry for account %s - this value is not '
+                         'intended to be manually created; generating new `token_salt`' % (APP_NAME, username))
+
+        # generate a new salt if the salt cannot be decoded or this is the initial run
+        if not decoded_salt:
+            decoded_salt = os.urandom(16)
+            token_salt = base64.b64encode(decoded_salt).decode('ascii')
+
+        # generate encrypter/decrypter based on password and random salt
+        key_derivation_function = PBKDF2HMAC(algorithm=hashes.SHA256(), length=32, salt=decoded_salt, iterations=100000,
```

Comment:
It looks like this is a bit low based on current recommendations. I believe this is not something that can be just updated and two different fernets probably need to be created and the old one can be used as a fallback when decrypting. A change like this is out of scope of this PR, and would require confirming a migration would be necessary.

Reply:
Yes – this is something I looked at improving a while ago, but it was hard to find a good way to migrate. Since it's a locally-stored encrypted value I felt the risk was quite low, but if you can suggest a way to do this that isn't onerous from the user side I'm happy to look at it.

Reply:
It looks like it might be relatively straightforward to do with a https://cryptography.io/en/latest/fernet/#cryptography.fernet.MultiFernet but you'd want to probably store the number of iterations used so that way you could run the migration and re-save the config when needed.